### PR TITLE
Added more test cases for TTL for Zero TTL and Created test cases for…

### PR DESCRIPTION
**Pull request by pubudup**
… topics. Updated README.me with test cases

### CODE REVIEW

**Review by nu1silva — COMMENTED** [review]
what are the uses of the following?
- mbtestcases.properties
- mbtestcases.xml
- module_mbtestcases.xml

I can build the project without them it seems.

**Review by nu1silva — CHANGES_REQUESTED** [review]
Check the other comments and if not required remove the files.

**Comment by pubudup** [comment]
Looks like these files were created with the idea project. I have removed these since it is not required to build the project after testing it.

**Review by nu1silva — APPROVED** [review]

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/DurableTopic/DurableTopicReceive.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package ttl.DurableTopic;
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/DurableTopic/DurableTopicReceive.java:2]: Missing license headers

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/DurableTopic/DurableTopicReceive.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            TextMessage message = (TextMessage) topicSubscriber.receive();
+
+            System.out.println("::Message Count::"+count+":::::::::::::Recieved message with content::::::::::::" + message.getText());
+
+            count++;
+
+        }
+        //queueReceiver.close();
+        // queueSession.close();
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/DurableTopic/DurableTopicReceive.java:60]: cleanup file

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/DurableTopic/DurableTopicSend.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package ttl.DurableTopic;
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/DurableTopic/DurableTopicSend.java:2]: Add license headers

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/DurableTopic/DurableTopicSend.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            // create the message to send
+
+            TextMessage message = topicSession.createTextMessage("Text Message with"+TTL+"::Published From IP::"+CARBON_DEFAULT_HOSTNAME);
+            //message.setStringProperty("msgID",TTL.toString());
+            javax.jms.TopicPublisher topicPublisher = topicSession.createPublisher(topic);
+            topicPublisher.publish(message);
+
+            topicPublisher.publish(message,DeliveryMode.PERSISTENT,4,10000);
+            topicPublisher.publish(message, DeliveryMode.PERSISTENT,4,40000);
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/DurableTopic/DurableTopicSend.java:57]: Clean up unwanted spaces

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/DifferentTTLSend.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
+package ttl.Queues;
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/DifferentTTLSend.java:2]: Add License

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/DifferentTTLSend.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import javax.jms.*;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import java.util.Properties;
+
+/**
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/DifferentTTLSend.java:9]: Provide proper class description

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/DifferentTTLSend.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    String queueName = "DefaultQueue";
+
+    public static void main(String[] args) throws NamingException, JMSException {
+        ZeroTTLSend queueSender = new ZeroTTLSend();
+        queueSender.sendMessages();
+    }
+
+    public void sendMessages() throws NamingException, JMSException {
+
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/DifferentTTLSend.java:33]: clean up space

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/DifferentTTLSend.java [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        QueueConnectionFactory connFactory = (QueueConnectionFactory) ctx.lookup(CF_NAME);
+        QueueConnection queueConnection = connFactory.createQueueConnection();
+        queueConnection.start();
+        QueueSession queueSession =  queueConnection.createQueueSession(false, QueueSession.AUTO_ACKNOWLEDGE );//
+
+        Queue queue = (Queue)ctx.lookup(queueName);
+
+        javax.jms.QueueSender queueSender = queueSession.createSender(queue);
+
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/DifferentTTLSend.java:58]: clean up unwanted spaces

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/QueueReceiveDefault.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
+package ttl.Queues;
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/QueueReceiveDefault.java:2]: Missing License headers

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/QueueReceiveDefault.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+            System.out.println("::Message Count::" + count + ":::::::::::::Recieved message with content::::::::::::" + message.getText());
+
+            count++;
+        }
+
+
+
+    }
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/QueueReceiveDefault.java:69]: Missing Doc comments

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/QueueSendDefault.java [hunk 11]
```diff
@@ -0,0 +1,91 @@
+package ttl.Queues;
+
```
Comment (dakshika) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/QueueSendDefault.java:2]: License missing

**Review by dakshika — COMMENTED** [review]
License missing
Needs to clean up unwanted spaces
Add Doc comments

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/QueueReceiveDefault.java [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import java.util.Properties;
+
+/**
+ * Created by pubudup on 9/5/16.
+ */
+public class QueueReceiveDefault {
+
+    public static final String QPID_ICF = "org.wso2.andes.jndi.PropertiesFileInitialContextFactory";
```
Comment (ruwany) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/Queues/QueueReceiveDefault.java:19]: Can we not use a constant file and retrieve these from one place. Configs seem to be repeated.

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/RedeliveryDelayWithTTL/Queues/ReceiveFromQueue.java [hunk 13]
```diff
@@ -0,0 +1,79 @@
+package ttl.RedeliveryDelayWithTTL.Queues;
```
Comment (ruwany) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/RedeliveryDelayWithTTL/Queues/ReceiveFromQueue.java:1]: Missing license.

File: products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/RedeliveryDelayWithTTL/Queues/ReceiveFromQueue.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+        queueConnection.start();
+
+        QueueSession queueSession =queueConnection.createQueueSession(false, Session.PER_MESSAGE_ACKNOWLEDGE );
+
+        Queue queue =  (Queue) ctx.lookup(queueName);
+        MessageConsumer queueReceiver =  queueSession.createConsumer(queue);
+        int count=1;
+
+        while(true) {
+
```
Comment (ruwany) [inline products/wso2_MB/combined_scenarios/MBTestCases/src/main/java/ttl/RedeliveryDelayWithTTL/Queues/ReceiveFromQueue.java:58]: Formatting issues?

**Review by ruwany — COMMENTED** [review]
Better to run through the files, once more and fix these issues wherever necessary.

**Comment by pubudup** [comment]
@dakshika @ruwany @ushanib : I have incorporated requested changes and sent a new pull request here: https://github.com/wso2/wso2-qa-artifacts/pull/127
